lib/tokenlist.cpp

```diff
@@ -1445,6 +1445,14 @@ static Token * findAstTop(Token *tok1, const Token *tok2)
 static Token * createAstAtToken(Token *tok, bool cpp)
 {
     // skip function pointer declaration
+    if (Token::Match(tok, "%type% * %var% [;=)]")) {
+        Token* vartok = tok->tokAt(2);
+        if (vartok && vartok->varId() > 0) {
```

Comment:
The `%var%` checks that the token is not null and that its varId is nonzero. So this condition at line 1450 duplicates the checks made in the Token::Match.